frontend/src/components/AppToggleExternalText.vue

```diff
@@ -1,179 +1,123 @@
-    toggleLeftText: {
+    leftText: {
       type: String,
-      default: "«Left«"
+      required: true
     },
-
-    /* Text of the false or right state */
-    toggleRightText: {
+    rightText: {
       type: String,
-      default: "»Right»"
+      required: true
     },
-
-    id: {
-      type: String,
-      default: "primary"
-    },
-
-    /* Default state of the toggle (false is left) */
-    defaultState: {
+    checked: {
       type: Boolean,
-      default: false
+      required: true
     }
   },
-
-  data() {
-    return {
-      currentToggleState: this.defaultState
-    };
-  },
-
-  watch: {
-    defaultState: function defaultState() {
-      this.currentToggleState = Boolean(this.defaultState);
-    }
-  },
-
-  computed: {
-    isActive() {
-      return this.currentToggleState;
-    },
-
-    /* Returns disabled / left text */
-    disabledText() {
-      return this.toggledLeftText;
-    },
-
-    /* Returns enabled / right text */
-    enabledText() {
-      return this.toggledRightText;
-    },
-
-    checkedValue: {
-      get() {
-        return this.currentToggleState;
-      },
-      set(newValue) {
-        this.currentToggleState = newValue;
-        this.$emit("change", newValue);
-      }
+  methods: {
+    update() {
+      this.$emit("update");
     }
   }
 };
 </script>
 
 <style scoped>
-/* TODO: Clean up styling */
-/* TODO: Grey out disabled toggle */
-/* TODO: Create constants for the accent colours for ease of modification */
-
-.toggle__label {
-  padding: 0 1rem;
-  font-size: 0.875em;
-}
-
-/* Hide Checkbox */
-.toggle__button input[type="checkbox"] {
-  display: none;
-}
-
-.toggle__button {
-  vertical-align: middle;
-  user-select: none;
+.container {
+  width: var(--switch-container-width);
   cursor: pointer;
+  display: flex;
+  align-items: center;
 }
-
-.toggle__button button[disabled] {
-  opacity: 0.5;
-  pointer-events: none;
+.left {
+  flex-shrink: 0;
+  margin-left: calc(var(--switch-container-width) * -1 + 2rem);
+  color: rgb(48, 48, 48);
 }
-
-/* Toggle Button Shape */
-
-.toggle__button .toggle__switch::after,
-.toggle__button .toggle__switch::before {
-  content: "";
-  position: absolute;
-  display: block;
-  height: 18px;
-  width: 18px;
-  border-radius: 50%;
-  left: 0;
-  top: -3px;
-  transform: translateX(0);
-  transition: all 0.25s cubic-bezier(0.5, -0.6, 0.5, 1.6);
+.leftWhite {
+  flex-shrink: 0;
+  margin-left: calc(var(--switch-container-width) * -1 + 2rem);
+  color: white;
 }
-
-.toggle__button .toggle__switch {
-  display: inline-block;
-  height: 12px;
-  border-radius: 6px;
-  width: 40px;
-  background: #b6caf7;
-  box-shadow: inset 0 0 1px #b6caf7;
-  position: relative;
-  margin-left: 10px;
-  transition: all 0.25s;
+.right {
+  flex-shrink: 0;
+  margin-left: 3rem;
+  color: rgb(48, 48, 48);
 }
-
-/* Toggle Left */
-.toggle__button .toggle__switch::after {
-  background: #6791f0;
-  box-shadow: 0 0 1px #666;
+.rightWhite {
```

Comment:
rename to `.rightOff` or `.rightDisabled`

Reply:
done!